tests/services/test_inference_timer.py

```
@@ -0,0 +1,90 @@
+    """Path-or-str is accepted; internal logging uses the basename."""
+    with patch("services.inference_timer.logger") as mock_logger:
+        with time_inference("text", str(tmp_path / "report.md")):
+            pass
+
+    _fmt, _kind, _ms, file_name = mock_logger.debug.call_args.args
+    assert file_name == "report.md"
+
+
+def test_zero_duration_is_not_negative() -> None:
+    """A no-op body still produces a non-negative reading."""
+    with time_inference("text", "/tmp/x") as t:
```

Ruff (0.15.14)

[error] Probable insecure usage of temporary file or directory: "/tmp/x"

(S108)